include/boost/numeric/ublas/eigen_solver.hpp

```diff
@@ -0,0 +1,395 @@
+				}
+				else {
+					std::complex<value_type> x(value_type(0), -T(k - size_type(1), k));
+					std::complex<value_type> y(T(k - size_type(1), k - size_type(1)) - p, q);
+					std::complex<value_type> cc = x / y;
+					T(k - size_type(1), k - size_type(1)) = cc.real();
+					T(k - size_type(1), k) = cc.imag();
+				}
+				T(k ,k - size_type(1)) = value_type(0);
+				T(k, k) = value_type(1);
+
+				for (size_type i = k - size_type(1); i-- != size_type(0);) {
+					
+					vector<value_type> ra_left = project(row(T, i), range(l, k + size_type(1))); 
+					vector<value_type> ra_right = project(column(T, k - size_type(1)), range(l, k + size_type(1)));
+					value_type ra = inner_prod(ra_left, ra_right);
+
+					vector<value_type> sa_right = project(column(T, k), range(l, k + size_type(1)));
+					value_type sa = inner_prod(ra_left, sa_right);
+
+					value_type w = T(i, i) - p;
+
+					if (eigenvalues_complex(i, i) < value_type(0)) {
+						lastw = w;
+						lastra = ra;
+						lastsa = sa;
+					}
+					else {
+						l = i;
+						if (eigenvalues_complex(i, i) == value_type(0)) {
+							std::complex<value_type> x(-ra, -sa);
+							std::complex<value_type> y(w, q);
+							std::complex<value_type> cc = x / y;
+							T(i, k - size_type(1)) = cc.real();
+							T(i, k) = cc.imag();
+						}
+						else {
+							value_type x = T(i, i + size_type(1));
+							value_type y = T(i + size_type(1), i);
+							value_type vr = (eigenvalues_real(i, i) - p) * (eigenvalues_real(i, i) - p) + eigenvalues_complex(i, i) * eigenvalues_complex(i, i) - q * q;
+							value_type vi = (eigenvalues_real(i, i) - p) * value_type(2) * q;
+
+							if (vr == value_type(0) && vi == value_type(0)) {
+								vr = (std::numeric_limits<value_type>::epsilon()) * norm * ((std::abs)(w)+(std::abs)(q)+(std::abs)(x)+(std::abs)(y)+(std::abs)(lastw));
+							}
+
+							std::complex<value_type> x1(x*lastra - lastw*ra + q*sa, x*lastsa - lastw*sa - q*ra);
+							std::complex<value_type> y1(vr, vi);
+							std::complex<value_type> cc = x1 / y1;
+							T(i, k - size_type(1)) = cc.real();
+							T(i, k) = cc.imag();
+
+							if ((std::abs)(x) > ((std::abs)(lastw)+(std::abs)(q))) {
+								T(i + size_type(1), k - size_type(1)) = (-ra - w * T(i, k - size_type(1)) + q * T(i, k)) / x;
+								T(i + size_type(1), k) = (-sa - w * T(i, k) - q * T(i, k - size_type(1))) / x;
+							}
+							else {
+								x1 = std::complex<value_type>(-lastra - y*T(i, k - size_type(1)), -lastsa - y*T(i, k));
+								y1 = std::complex<value_type>(lastw, q);
+								cc = x1 / y1;
+								T(i + size_type(1), k - size_type(1)) = cc.real();
+								T(i + size_type(1), k) = cc.imag();
+							}
+
+						}
+
+					}
+
+					value_type t = (std::max)((std::abs)(T(i, k - size_type(1))), (std::abs)(T(i, k)));
+					if ((std::numeric_limits<value_type>::epsilon() * t)*t > value_type(1)) {
+						for (size_type j = i; j < k + i - size_type(1); j++){
```

Comment:
The bound of `k + i - size_type(1)` here is incorrect and will cause `j` to index past the end of the matrix. Compare with [EigenSolver.h](https://eigen.tuxfamily.org/dox/EigenSolver_8h_source.html) lines 596-598.
See [thomasyang1207 pull request #5](https://github.com/thomasyang1207/ublas/pull/5) for the fix.